text/builders/0000-jammy-builders.md

```diff
@@ -0,0 +1,150 @@
+# Builders based on Ubuntu 2022.04: Jammy Jellyfish
+
+## Summary
+
+A set of stacks based on the Ubuntu 2022.04 LTS (Jammy Jellyfish) will be
+maintained by Paketo, per stacks [RFC
+0004](https://github.com/paketo-buildpacks/rfcs/blob/da3339d071ffed23c3cd1b374a6bfcefdea7ac70/text/stacks/0004-jammy-jellyfish.md).
+Builders are an important way for Paketo users to consume stacks; they can be
+used as an input to the `pack` CLI. Paketo should maintain a set of builders
+that use the Jammy Jellyfish stacks.  They should be developed, released, and
+maintained by the Builders subteam.  These new builders should come in "full",
+"buildpackless-full", "base", "buildpackless-base",  "tiny", and
+"buildpackless-tiny" variants.
+
+## Motivation
+
+Paketo [plans to support Jammy Jellyfish
+stacks](https://github.com/paketo-buildpacks/rfcs/blob/da3339d071ffed23c3cd1b374a6bfcefdea7ac70/text/stacks/0004-jammy-jellyfish.md).
+Paketo [builders](https://paketo.io/docs/concepts/builders/) are a common way
+for Paketo users to consume Paketo stacks, buildpacks, and the Cloud Native
+Buildpacks (CNB) lifecycle. In order for many Paketo users to get value from
+Jammy Jellyfish stacks, we'll need to provide Jammy Jellyfish builders.
+
+## Detailed Explanation
+
+### Variants
+
+The Jammy builders will be available in 6 variants that align to the existing
+Bionic-based builders. These will be called:
+- `jammy-full`
+- `jammy-buildpackless-full`
+- `jammy-base`
+- `jammy-buildpackless-base`
+- `jammy-tiny`
+- `jammy-buildpackless-tiny`
+
+The Jammy builders will eventually* contain corresponding sets of buildpacks to
+their Bionic counterparts. Buildpackless builders will contain no buildpacks,
+as described in the [Buildpackless Builders
+RFC](https://github.com/paketo-buildpacks/rfcs/blob/da3339d071ffed23c3cd1b374a6bfcefdea7ac70/text/0030-buildpackless-builders.md).
+
+\*Note: Some buildpacks (or the tools they install) may not yet be compatible
+with the Jammy Jellyfish operating system. For instance, this is an [open
+discussion](https://github.com/dotnet/core/issues/7038) for .NET Core. While
+the Jammy builders should mirror existing Paketo builders as best as possible,
+if buildpack or language support presents a blocker, it is acceptable to
+release builders with best-effort subsets of the buildpacks supported by the
+Bionic builders.
+
+### Image Naming and Tagging
+
+The stacks will name and tag their release images with the following pattern:
+
+```
+paketobuildpacks/builder-jammy-{variant}:{version}
```

Comment:
Unless... maybe the `pack builder create` command is capable of building a builder image [manifest list](https://docs.docker.com/engine/reference/commandline/manifest/) that contains images of different architectures. This is worth investigating. I'll take an action item.

Reply:
I think that the `jam create-stack` is capable of having a manifest list which I think is maybe the only thing that matters?

Reply:
It seems like from the conversation in CNB Slack, that the manifest list idea is viable/well received but could possibly be coupled with the removal of the stacks/mixins concept. The removal of stacks will lead to other changes we need to make most likely (what will we do with the stacks/variants we publish now?). If manifest support does get rolled into those CNB changes, we can make the necessary changes in Paketo when _everything_ is available in CNB. 

My thought is that we should support multiple builder variants (`paketobuildpacks/builder-{os-flavor}-{architecture}-{variant}:{version}` as proposed) since we do the same for stacks for now, with the intent to push forward a manifest list option in pack down the road. @fg-j thoughts?

Reply:
Question, even if pack does support the manifest list idea, won't we still have a naming conundrum? Or would the output of `pack builder create` be a single, multi-arch image?

Reply:
100% agree with the direction this thread is heading in. We should not be baking the architecture into a tag. Let the image manifest and image fetching code handle that negotiation.

Reply:
Addressed in most recent commit https://github.com/paketo-buildpacks/rfcs/pull/202/commits/3db5e82a8ac25adc9e384b8b6963224bd2d44b99